src/aws-cpp-sdk-core/include/aws/core/utils/crypto/crt/CRTHash.h

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/core/utils/crypto/MD5.h>
+#include <aws/core/utils/crypto/Sha1.h>
+#include <aws/core/utils/crypto/Sha256.h>
+#include <aws/core/utils/crypto/Sha256HMAC.h>
+
+#include <aws/crt/crypto/Hash.h>
+
+namespace Aws
+{
+    namespace Utils
+    {
+        namespace Crypto
+        {
+            class AWS_CORE_API CRTHash : public Hash
+            {
+            public:
+                ~CRTHash() override = default;
+                CRTHash(const CRTHash &) = delete;
+                CRTHash &operator=(const CRTHash &) = delete;
+                CRTHash(CRTHash &&) = default;
+                CRTHash &operator=(CRTHash &&) = default;
+
+                HashResult Calculate(const String &str) override;
```

Comment:
oof on string? that can have a lot of "fun" corner cases. is it possible to switch to a byte buf or its too late now?

Reply:
too late now, that is part of the interface for [hash that has long existed](https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-core/include/aws/core/utils/crypto/Hash.h#L29-L32)

Reply:
Please add a todo to deprecate it, possibly a deprecation macro warning, but likely just a todo.